### ISSUE TITLE
🐛 [Firebase_messaging]  duplication bug and two threads launched by background handler

### ISSUE THREAD

**Issue by bibaswan-bhawal** [firebase/flutterfire#9912]
## Firebase_messaging duplication bug report

**Describe the bug**
FirebaseMessaging.onMessage.listen being called twice for one notification.

### Steps to reproduce

Run example firebase_messaging app with news libraries and send a notification while app in foreground

4. See error or incorrect behavior

I/flutter (18638): Receive a notification
I/flutter (18638): Receive a notification
W/FirebaseMessaging(18638): Unable to log event: analytics library is missing
W/FirebaseMessaging(18638): Unable to log event: analytics library is missing

### Expected behavior

I/flutter (18638): Receive a notification
W/FirebaseMessaging(18638): Unable to log event: analytics library is missing

### Sample project

example firebase_messaging

**Comment by turbica** [comment]
Same problem here.

Rolling back to version 14.0.4 seemed to solve the problem.

In this case, the use of the 'FlutterEngineGroup' in the version 14.1.0 and 'FirebaseMessaging.onBackgroundMessage' cretes a new isolated instance that force all listenable events in the app to execute twice. ALL listenable events, which is a MADNESS.

**Comment by mAlkhatieb** [comment]
same problem

**Comment by masewo** [comment]
As @turbica has already explained, version `14.1.0` breaks a lot of other dependencies in my app, like `flutter_displaymode` and `flutter_web_auth`, because of the `FlutterEngineGroup` change. Plugin callbacks are called twice, but the second time without an activity.

**Comment by darshankawar** [comment]
@bibaswan-bhawal and everyone facing this issue: Is this only happening on Android or iOS too ?
I checked the `firebase_messaging` plugin example using latest version, sent a notification with app in foreground as well as in background, and the console only showed it once as expected:

```
flutter: A new onMessage event was published!

flutter: Handling a background message 1668423323857618

```

Can anyone please confirm if you are also seeing this using plugin example too ?

**Comment by madushans** [comment]
using the latest versions of firebase causes ` FirebaseMessaging.onBackgroundMessage(...);` to spin up a duplicate isolate and causes the main method to run twice, which causes multiple registrations for handlers, the build methods running twice .etc.

using the below versions together seems to work without the issue.

```
  firebase_core: 1.24.0
  firebase_remote_config: 2.0.20
  firebase_analytics: 9.3.8
  firebase_crashlytics: 2.9.0
  firebase_messaging: 13.1.0
```

see https://stackoverflow.com/q/74430484/975887


To repro the issue, use versions below

```
firebase_core: ^2.2.0 
firebase_messaging: ^14.1.0
```

**Comment by bibaswan-bhawal** [comment]
@darshankawar This issue has only been observed on Android, works as expected on IOS.

**Comment by darshankawar** [comment]
Thanks for the info. Like I tried [here](https://github.com/firebase/flutterfire/issues/9932#issuecomment-1316521832), I don't seem to get duplicated logs, but based on the report and comments like https://github.com/firebase/flutterfire/issues/9912#issuecomment-1312541760, keeping this issue open and labeling for further insights and tracking for team.

Maybe related: https://github.com/firebase/flutterfire/issues/9932

**Comment by Lyokone** [comment]
Thanks for all the reports, it'll help track down the exact reason with more precision, I have pushed a PR to revert the change, and will investigate more on it 👍

**Comment by SamerOrfali** [comment]
it's solved in version firebase_messaging: ^14.1.1

**Comment by Lyokone** [comment]
After discussing with the Flutter team, we have decided to go for another solution to reduce performance hit and wait for the next stable release. Sorry for the inconvenience.
